cluster-lifecycle/update-managedcluster-status-to-include-clusterinfo.md

```diff
@@ -0,0 +1,254 @@
+status: provisional
+---
+
+# Update ManagedCluster status to include ClusterInfo
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The proposed work would integrate the ClusterInfoStatus into ManagedCluster:
+
+https://github.com/open-cluster-management/multicloud-operators-foundation/blob/master/pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go#L22-L62
+
+## Motivation
+
+Today, several label properties, including clusterID, cloud and vendor, are updated on ManagedCluster objects that can be modified by users, creating invalid data for matching behavior. It would be a good idea to include cluster info provided by the Klusterlet in the status of ManagedCluster.
+
+It is required from other components that more cluster specific info should be added into ManagedCluster API. This information includes the clusterID, the ocp distribution version, vendor and cloud provider etc. On another hand, ManagedCluster API should be designed to have a generic and extensible structure to include this info that also can work with different implementations.
+
+### Goals
+
+- Allow extensible info in managed cluster to be added in ManagedCluster API
+- Remove cluster info lables from managed cluster eventually and use managed cluster status instead;
+- Align wth upstream design;
+
+### Non-Goals
+
+- Replace and deprecate ManagedClusterInfo API
+
+## Proposal
+
+To align with the upstream proposal for ClusterID (https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit), leverage ClusterClaims created on managed cluster to store and manage cluster information.
+
+### Risks and Mitigations
+
+- The upstream proposal for ClusterID is not finalized yet and may change in the future.  
+The mitigation is to create ClusterClaim CRD in group cluster.open-cluster-management.io with version v1alph1 for now. Once the proposal is finalized, support both of them for one or two releases and then deprecate the one in group cluster.open-cluster-management.io.
+
+## Design Details
+
+As shown in the table below, ManagedClusterInfo includes 10 fields in its status. Some of them will NOT be merged into ManagedCluster.
+
+| Name    | Description    | Comments   |
+| ---- | ---- | ---- |
+| Conditions | condition information for a managed cluster | **Not merged**. ManagedCluster has its own Conditions. |
+| Version | the kube version of managed cluster | **Not merged**. ManagedCluster has a field with same name in its status, which also points to the kube version of managed cluster |
+| KubeVendor | the kubernetes provider of the managed cluster| |
+| CloudVendor | the cloud provider for the managed cluster | |
```

Comment:
Is this more of a `Platform`?  so I would have tuples like
1. ocp NA, kube 1.18, gke
2. ocp 4.6, kube 1.19, aws
3. ocp 4.6, kube 1.19, ovirt

Reply:
Assuming you meant 'gce' here.

I'm also in favor of `CloudPlatform` as the name for this field.

A couple scenarios to think about here:

- How would an OCP cluster on AWS look vs an EKS cluster?
- How would an OCP cluster on GCE look vs a GKE cluster?
- How would an OCP cluster look on equinix metal vs my own hardware?

Reply:
How about we create a ClusterClaim with name `CloudPlatform`? Its value has the format below:

`<kube_version>,<distribution_name>,<distribution_version>,<platform>`

Here are some examples.

| Value                            | Description                     |
| -------------------------------- | ------------------------------- |
| 1.19,OpenShift,4.6,AWS           | An OCP cluster on aws           |
| 1.19,NA,NA,EKS                       | An EKS cluster                  |
| 1.19,OpenShift,4.6,GCE           | An OCP cluster on GCE           |
| 1.19,NA,NA,GKE                       | An GKE cluster                  |
| 1.19,OpenShift,4.6,Equinix Metal | An OCP cluster on Equinix Metal |
| 1.19,OpenShift,4.6,oVirt         | An OCP cluster on oVirt         |
| 1.19,OpenShift,4.6,NA              | An OCP cluster on own hardware  |

Reply:
I'd rather have distinct fields for distinct pieces of info than have to parse fields, how about you @deads2k

Reply:
> I'd rather have distinct fields for distinct pieces of info than have to parse fields, how about you @deads2k

I agree about making distinct claims.  Based on the ones listed above, it looks like we want
1. kubernetes-version
2. ocp-version
3. platform
4. (maybe?) product - this would be something like openshift, anthos, eks, gke, etc

Reply:
Replaced CloudPlatform with 4 distinct claims:
- kubeversion.cluster.open-cluster-management.io
- platform.cluster.open-cluster-management.io
- product.cluster.open-cluster-management.io
- version.openshift.io